sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/HyperLogLogPlusPlus.scala

```diff
@@ -135,7 +135,7 @@ case class HyperLogLogPlusPlus(
 
   override def dataType: DataType = LongType
 
-  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType, DoubleType)
```

Comment:
HyperLogLogPlusPlus was failing because I was passing it Decimal literals. I thought I could solve this by casting. While this is not relevant anymore, I still think this is valid.